src/network.ts

```diff
@@ -153,7 +154,29 @@ export class Request {
     return this._postData;
   }
 
-  headers(): {[key: string]: string} {
+  postDataJSON(): Object | null {
+    if (!this._postData)
+      return null;
+
+    const contentType = this.headers()['content-type'];
+    if (!contentType)
+      return null;
+
+    if (contentType === 'application/x-www-form-urlencoded') {
+      const entries: Record<string, string> = {};
+      const parsed = new URLSearchParams(this._postData);
+      for (const [k, v] of parsed.entries())
+        entries[k] = v;
+      return entries;
+    }
+
+    if (contentType.startsWith('application/json'))
```

Comment:
I wonder if we should `JSON.parse` unconditionally at this point. Users will post w/o proper mime type and would expect `postDataJSON` to work. It'll throw if data is a not a JSON string, which is Ok.